sh/tests/comparator_test.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# Unofficial Bash Strict Mode
+set -euo pipefail
+IFS='
+'
+
+# Test if there were only two arguments given
+
+script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
+
+challenge() {
+    if [ $# -ne 2 ]; then
+        submitted=$(bash "$script_dirS"/student/comparator.sh $1)
+        expected=$(bash "$script_dirS"/solutions/comparator.sh $1)
+    else
+        submitted=$(bash "$script_dirS"/student/comparator.sh $1 $2)
+        expected=$(bash "$script_dirS"/solutions/comparator.sh $1 $2)
+        diff <(echo "$submitted") <(echo "$expected")
```

Comment:
the `diff` should be after the `if`- the test is not failing when the number of argument is different then 2 and there are differences